source/lib/position/index.js

```diff
@@ -0,0 +1,167 @@
+import { yOffsetFnMap, xOffsetFnMap } from './offsetFunctions';
+
+/**
+ * Calculate the top, right, bottom, and left x/y values of
+ * an element at given coordinates.
+ *
+ * @function
+ * @param {HTMLElement} element
+ * @param {Object} coord - (x,y) coordinates
+ */
+function _getElementBox (element, coord) {
+    let boundingRect = element.getBoundingClientRect();
+
+    return {
+        top: coord.y,
+        right: coord.x + boundingRect.width,
+        bottom: coord.y + boundingRect.height,
+        left: coord.x,
+    };
+}
+
+/**
+ * Calculate coordinates of an element in relation to a reference element.
+ *
+ * @function
+ * @param {String} position - the position of the offset element
+ * @param {HTMLElement} offsetElement - the element to calculate (x,y) coordinates
+ * @param {HTMLElement} referenceElement - the element that is being offset from
+ *
+ * @returns {Object} absolute (x,y) coordinates and metadata to position offsetElement
+ * in relation to referenceElement
+ */
+function _getCoords (position, offsetElement, referenceElement) {
+    // The 'position' property is added to provide information about final
+    // calculated position of offset element in relation to reference element
+    let coords = {
+        x: 0,
+        y: 0,
+        position,
+    };
+    let offRect = offsetElement.getBoundingClientRect();
+    let refRect = referenceElement.getBoundingClientRect();
+
+    coords.x = xOffsetFnMap[position](offRect, refRect);
+    coords.x += window.pageXOffset;
+
+    coords.y = yOffsetFnMap[position](offRect, refRect);
+    coords.y += window.pageYOffset;
+
+    return coords;
+}
+
+/**
+ * Determine if any side of an element is obscured by the viewport.
+ *
+ * @function
+ * @argument {HTMLElement} element - the element to check against the viewport
+ * @argument {Object} coords - (x,y) coordinates
+ *
+ * @returns {Object} metadata object with boolean values to quickly
+ * identify which sides of an element are outside the viewport
+ */
+function _getOffscreenMetadata (element, coords) {
+    let elementBox = _getElementBox(element, coords);
+    let viewportBox = {
+        top: window.pageYOffset,
+        right: window.innerWidth + window.pageXOffset,
+        bottom: window.innerHeight + window.pageYOffset,
+        left: window.pageXOffset,
+    };
+
+    let offscreen = {
+        top: elementBox.top < viewportBox.top,
+        right: elementBox.right > viewportBox.right,
+        bottom: elementBox.bottom > viewportBox.bottom,
+        left: elementBox.left > viewportBox.left,
```

Comment:
This got flipped the wrong way somehow.

`left: elementBox.left < viewportBox.left,`

Reply:
ah, good catch. will fix